pystarport/expansion.py

```diff
@@ -42,29 +41,22 @@ def _expand(value, variables):
     return "".join([str(atom.resolve(variables)) for atom in atoms])
 
 
-def expand_yaml(config_path, dotenv):
+def expand_yaml(config_path, dotenv_from_param):
     config = yaml.safe_load(open(config_path))
-
-    def expand(dotenv):
-        if not isinstance(dotenv, str):
-            raise ValueError(f"Invalid value passed to dotenv: {dotenv}")
-        config_vars = dict(os.environ)  # load system env
-        env_path = Path(config_path).parent.joinpath(dotenv)
-        if not env_path.is_file():
-            raise ValueError(
-                f"Dotenv specified in config but not found at path: {env_path}"
-            )
-        config_vars.update(dotenv_values(dotenv_path=env_path))  # type: ignore
-        load_dotenv(dotenv_path=env_path)
-        return expand_posix_vars(config, config_vars)
-
-    if dotenv is not None:
-        if "dotenv" in config:
-            _ = config.pop("dotenv", {})  # remove dotenv field if exists
-        dotenv_path = dotenv
-        config = expand(dotenv_path)
-    elif "dotenv" in config:
-        dotenv_path = config.pop("dotenv", {})  # pop dotenv field if exists
-        config = expand(dotenv_path)
-
+    dotenv_from_config = config.pop("dotenv", {})
+    merged = {}
+    parent = Path(config_path).parent
+    for d in [dotenv_from_config, dotenv_from_param]:
+        if d not in (None, '', {}):
+            env_path = parent.joinpath(d)
+            if not env_path.is_file():
+                raise ValueError(
+                    f"Dotenv specified in config but not found at path: {env_path}"
+                )
+            merged = {
```

Comment:
I think this approach is not intuitive from user's point of view.

Reply:
just think dotenv merge is easier than yaml, but might need wait [merge](https://github.com/mmsqe/mergedeep/commit/8be9c7b5bc9403ac0a86cc281a7644f49363ce38) change get merged